core/modules/path/path.module

```diff
@@ -22,11 +48,29 @@ function path_permission() {
     'notify of path changes' => array(
       'title' => t('Notify of automatic path changes'),
       'description' => t('Determines whether or not users are shown a notice when an automatic alias changes.'),
-    )
+    ),
+    'administer redirects' => array(
+      'title' => t('Administer URL redirections'),
+    ),
   );
 }
 
 /**
+ * Implements hook_help().
+ */
+function path_help($path, $arg) {
+  $output = '';
+  switch ($path) {
+    case 'admin/config/search/redirect/404':
+      $output = '<p>' . t('This page lists all paths that have resulted in 404 errors and do not yet have any redirects assigned to them.') . '</p>';
```

Comment:
If we want to keep this help text, move it to the top of the form as `$form['help'] = array('#markup' => '...')`